recipes/nmos-cpp/all/conanfile.py

```diff
@@ -73,12 +73,15 @@ def requirements(self):
 
     def build_requirements(self):
         # nmos-cpp needs CMake 3.17 or higher but CCI doesn't allow version ranges
-        self.build_requires("cmake/3.21.2")
+        self.build_requires("cmake/3.22.0")
 
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
             tools.check_min_cppstd(self, 11)
 
+    def package_id(self):
+        self.info.requires["boost"].minor_mode()
```

Comment:
I'm wondering: do we also want this mode if nmos-cpp is static?

Reply:
nmos-cpp does not support shared mode. If nmos-cpp is built against Boost 1.X, recipes using nmos-cpp and Boost must absolutely link against the same Boost 1.X, same as the case for cpprestsdk.

Reply:
Indeed, there is no shared option.
So for my own understanding: which components of boost are responsible of this? It is still true if boost is static?

Reply:
Boost.ASIO was the culprit between Boost 1.76.0 and Boost 1.77.0 I believe.

Reply:
I don't know which specific components cause that problem - maybe @garethsb does - but that is irrespective of static or shared. The problem is that the ABI changed, so one module would generate e.g. a certain memory layout for a piece of data, while the other component was expecting a different layout.

Reply:
That would makes sense, it was a huge update https://github.com/boostorg/asio/compare/boost-1.76.0...boost-1.77.0#diff-6e6a90601914ff7c360a3ae54062f6c22ee7ffc75c1102b8e6980a25afc904eaR10